Bottom: 7e169d65063c24862413a8436f7fbacdf4ed2a2d
Top:    ba22513edc0acedc969909b52b58524e8d5bca9e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-11 17:01:21 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index dbb3d12..c897667 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -236,6 +236,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
